Fix: actually run integration tests

ilmotta opened the pull request:
Fixes https://github.com/status-im/status-mobile/issues/17895

### Summary

Integration tests are not running anymore since https://github.com/status-im/status-mobile/pull/17762 was merged one week ago.

This PR also allows individual integration test namespaces to run if you change the `:ns-regexp` option in shadow-cljs. This is quite handy, since they are relatively slow.

#### Areas that may be impacted

None.

### Steps to test

The CI is already testing for us, just `make test`.

status: ready

ilmotta committed:
Fix: run integration tests

Also allow each integration test to run independently if filtered by shadow-cljs
option :ns-regexp.

ilmotta committed:
Use qualified symbols in macros to help debug

ilmotta committed:
Fix: exception thrown due to reference to var instead of keyword

ilmotta committed:
Take advantage of idempotency in utils.test/init!

ilmotta commented on src/status_im/utils/test.cljs:
```diff
@@ -15,17 +15,21 @@
-  (js/setInterval (fn []
-                    (.pollSignal native-status signal-received-callback)
-                    100)))
+  (when-not @initialized?
```
FYI: Here we make `init!` idempotent, so we can safely and automatically call it for every test using the macro `with-app-initialized`. This is what allows each namespace to run independently and in any order.

ilmotta commented on src/test_helpers/integration.cljs:
```diff
@@ -1,4 +1,5 @@
 (ns test-helpers.integration
+  (:require-macros [test-helpers.integration])
```
FYI: Macros weren't loaded in `develop`, so nothing really worked.

mohsen-ghafouri reviewed (approved):
🚀

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | bdfee66f | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-17900/1/) | 2023-11-14 14:35:17 | ~6 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-231114-142838-bdfee6-pr17900-universal.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-231114-142838-bdfee6-pr17900-universal.apk)|

ilmotta committed:
Skip wallet integration tests

ilmotta commented:
Reviewers: @cammellos, @J-Son89, @mohsen-ghafouri, @briansztamfater

I just rebased and found that the wallet integration tests started to fail consistently. The culprit is probably the recent refactors made yesterday by commit 45df308dd0d0e2b3945a70f514dcf5fc5f596abe.

Unfortunately I couldn't get this PR merged before due to CI flakiness, hence I just moved on to other tasks.

Given that the scope of this PR is to re-enable integration tests and allow them to run independently, I just skipped the wallet integration tests to get this PR in. I'm going to create an issue to fix/re-implement them, so someone with more context can fix them.

cc @ulisesmac

ulisesmac commented:
> Reviewers: @cammellos, @J-Son89, @mohsen-ghafouri, @briansztamfater
> 
> I just rebased and found that the wallet integration tests started to fail consistently. The culprit is probably the recent refactors made yesterday by commit [45df308](https://github.com/status-im/status-mobile/commit/45df308dd0d0e2b3945a70f514dcf5fc5f596abe).
> 
> Unfortunately I couldn't get this PR merged before due to CI flakiness, hence I just moved on to other tasks.
> 
> Given that the scope of this PR is to re-enable integration tests and allow them to run independently, I just skipped the wallet integration tests to get this PR in. I'm going to create an issue to fix/re-implement them, so someone with more context can fix them.
> 
> Edit: issue to fix wallet integration tests created #17921
> 
> cc @ulisesmac

@ilmotta were those tests passing before you rebased?

I'm taking a look at the ns `status-im2.integration-test.wallet`, but I didn't touch any of the involved subscriptions :thinking: so I'm not sure about the root cause. 
What is the commit that disabled the integration tests?
